File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/tourbooking/event/BookingActivated.java
package com.dominikgaller.alpinebooking.booking.core.domain.tourbooking.event;

import com.dominikgaller.alpinebooking.shared.domain.event.DomainEvent;
import com.dominikgaller.alpinebooking.booking.core.domain.tourbooking.BookingId;

import java.time.Instant;

/**
 * Domain event emitted when a tour booking is marked active.
 *
 * <p>Immutable. Published after successful transaction commit.
 * {@code guideTourId} is an optional correlation identifier linking the booking
 * to the guide tour execution that triggered the activation.
 *
 * <p>{@code guideTourId} <b>may be null</b> — a caller without a correlation id may still
 * activate a booking. Nullable rather than {@link java.util.Optional} because a record
 * component is a field, not a query ({@code coding-style.definition.md} § 1.4 exception).
 *
 * <p>It is a plain {@link String} by design — the identity is owned by the {@code guide}
 * context and is opaque here (ADR-0005 category 2).
 *
 * <p>SDD: See {@code documentation/use-cases/uc06-mark-booking-active.spec.md}
 *          and {@code documentation/adr/0005-bounded-context-identity-boundaries.adr.md}.
 */
public record BookingActivated(
        BookingId bookingId,
        Instant activatedAt,
        String guideTourId
) implements DomainEvent {
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/event/BookingActivated.kt
package com.example.contractmanagement.booking.core.domain.tourbooking.event

import com.example.contractmanagement.booking.core.domain.tourbooking.BookingId
import com.example.contractmanagement.shared.domain.event.DomainEvent
import java.time.Instant

/**
 * A confirmed booking became active because its tour started (UC06).
 *
 * @property guideTourId correlation id of the guide tour execution that caused this.
 *   Null when the activation had no guide tour to correlate with. Deliberately a plain
 *   [String]: the identity is owned by the `guide` context, so `booking` treats it as
 *   opaque and never parses or branches on it
 *   (`documentation/adr/0005-bounded-context-identity-boundaries.adr.md`).
 *
 * SDD: see `documentation/use-cases/uc06-mark-booking-active.spec.md` § 6.
 */
data class BookingActivated(
    val bookingId: BookingId,
    val activatedAt: Instant,
    val guideTourId: String?,
) : DomainEvent
